src/Sylius/Bundle/TaxonomyBundle/Doctrine/ORM/TaxonRepository.php

```diff
@@ -12,27 +12,37 @@
 namespace Sylius\Bundle\TaxonomyBundle\Doctrine\ORM;
 
 use Sylius\Bundle\TranslationBundle\Doctrine\ORM\TranslatableResourceRepository;
-use Sylius\Component\Taxonomy\Model\TaxonomyInterface;
+use Sylius\Component\Taxonomy\Model\TaxonInterface;
 use Sylius\Component\Taxonomy\Repository\TaxonRepositoryInterface;
 
 /**
- * @author Saša Stamenković <umpirsky@gmail.com>
- * @author Gonzalo Vilaseca <gvilaseca@reiss.co.uk>
- * @author Anna Walasek <anna.walasek@lakion.com>
+ * @author Aram Alipoor <aram.alipoor@gmail.com>
  */
 class TaxonRepository extends TranslatableResourceRepository implements TaxonRepositoryInterface
 {
-    public function getTaxonsAsList(TaxonomyInterface $taxonomy)
+    /**
+     * @param TaxonInterface $taxon
+     *
+     * @return array
+     */
+    public function findChildren(TaxonInterface $taxon)
     {
-        return $this->getQueryBuilder()
-            ->where('o.taxonomy = :taxonomy')
-            ->andWhere('o.parent IS NOT NULL')
-            ->setParameter('taxonomy', $taxonomy)
-            ->orderBy('o.left')
-            ->getQuery()
-            ->getResult();
+        $queryBuilder = $this->getQueryBuilder();
+        $queryBuilder
+            ->andWhere(':left < '.$this->getAlias().'.left AND '.$this->getAlias().'.right < :right')
```

Comment:
How about:
```php
    ->andWhere($queryBuilder->expr()->gt('o.left', ':left'))
    ->andWhere($queryBuilder->expr()->lt('o.right', ':right'))
```

We're not going to use the `getAlias` method anymore and always write `o` as an alias (see https://github.com/Sylius/Sylius/pull/3854).